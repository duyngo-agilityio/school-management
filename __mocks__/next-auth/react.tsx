import { createContext } from "react";

export const useSession = () => ({
  data: {
    user: {
      token: "access-token",
      email: "user@gmail.com",
      id: "1",
      name: "User",
    },
    expires: Date.now().toString(),
  },
  status: "authenticated",
  update: async () => null,
});

export type Session = {
  data: {
    user: {
      token: string;
      email: string;
      id: string;
      name: string;
    };
    expires: string;
  };
  status: "loading" | "authenticated" | "unauthenticated";
  update: (data?: Session) => Promise<Session | null>;
};

const INITIAL_SESSION: Session = {
  data: {
    user: {
      token: "access-token",
      email: "user@gmail.com",
      id: "1",
      name: "User",
    },
    expires: Date.now().toString(),
  },
  status: "authenticated",
  update: async () => null,
};

const SessionContext = createContext(INITIAL_SESSION);

export const SessionProvider = ({ children }: React.PropsWithChildren) => {
  <SessionContext.Provider value={INITIAL_SESSION}>
    {children}
  </SessionContext.Provider>;
};
