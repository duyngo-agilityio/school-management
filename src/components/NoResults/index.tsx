import Image from 'next/image';
import Link from 'next/link';

// Components
import { Flex, Text } from '@chakra-ui/react';

// Constants
import { PAGE_NOT_FOUND_IMAGE } from '@/constants';

interface NoResultsProps {
  title: string;
  src?: string;
  description?: string;
  backButton?: string;
  href?: string;
  width?: number;
  height?: number;
}

const NoResults = ({
  src = PAGE_NOT_FOUND_IMAGE,
  title,
  description,
  backButton,
  href = '',
  width = 340,
  height = 225,
}: NoResultsProps) => {
  return (
    <Flex flexDirection="column" justifyContent="center" alignItems="center">
      <Image src={src} alt="" width={width} height={height} />
      <Text variant="primary" fontWeight="semibold" size="xl" mt="31px">
        {title}
      </Text>
      <Text variant="primary" size="sm" mt="4px">
        {description}
      </Text>
      <Link href={href}>{backButton}</Link>
    </Flex>
  );
};

export default NoResults;
