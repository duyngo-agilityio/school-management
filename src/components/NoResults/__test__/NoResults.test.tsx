import { render } from '@testing-library/react';

// Components
import NoResults from '..';

const NoResultsProps = {
  title: 'Page Not Found',
  src: '/no-notification.png',
  description: 'Back home page',
  width: 340,
  height: 225,
};

describe('NoResults Component', () => {
  it('should match snapshot for NoResults', () => {
    const { container } = render(<NoResults {...NoResultsProps} />);

    expect(container).toMatchSnapshot();
  });
});
