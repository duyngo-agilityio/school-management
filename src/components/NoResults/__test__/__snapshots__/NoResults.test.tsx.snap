// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResults Component should match snapshot for NoResults 1`] = `
<div>
  <div
    class="css-wkpw2c"
  >
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      height="225"
      loading="lazy"
      src="/_next/image?url=%2Fno-notification.png&w=750&q=75"
      srcset="/_next/image?url=%2Fno-notification.png&w=384&q=75 1x, /_next/image?url=%2Fno-notification.png&w=750&q=75 2x"
      style="color: transparent;"
      width="340"
    />
    <p
      class="chakra-text css-1uy2b59"
    >
      Page Not Found
    </p>
    <p
      class="chakra-text css-1fhgjcy"
    >
      Back home page
    </p>
    <a
      href=""
    />
  </div>
</div>
`;
